lib/client.c

```diff
@@ -306,18 +309,34 @@ static sf_bool STDCALL log_init(const char *log_path, SF_LOG_LEVEL log_level) {
     SF_LOG_LEVEL sf_log_level = log_level;
     char strerror_buf[SF_ERROR_BUFSIZE];
 
+    char client_config_file[MAX_PATH] = {0};
+    snowflake_global_get_attribute(
+      SF_GLOBAL_CLIENT_CONFIG_FILE, client_config_file, sizeof(client_config_file));
+    client_config clientConfig = { .logLevel = "", .logPath = "" };
```

Comment:
memory leak currently with the buffer allocated in load_client_config().
When you try to free the buffer, initialize with const string could be a problem.